pandas/core/arrays/string_.py

```diff
@@ -152,13 +157,13 @@ class StringArray(PandasArray):
     ['This is', 'some text', <NA>, 'data.']
     Length: 4, dtype: string
 
-    Unlike ``object`` dtype arrays, ``StringArray`` doesn't allow non-string
-    values.
```

Comment:
I think it is important to keep the original intent of this part, in some way. As it was meant to explain that the StringArray will only *contain* strings (as opposed to object dtype, which is not strict here). This could maybe also shown with setting a non-string value (`arr[0] = 1`) or so.

Reply:
How you changed it still doesn't explain the original intent, as mentioned above, IMO. I would explain both (only strings allowed + showing the automatic conversion)

Reply:
Ok, I've tried it differently.